cvat-ui/src/components/annotation-page/canvas/views/canvas2d/canvas-wrapper.tsx

```diff
@@ -597,6 +608,12 @@ class CanvasWrapperComponent extends React.PureComponent<Props> {
         canvasInstance.html().removeEventListener('canvas.message', this.onCanvasMessage as EventListener);
     }
 
+    // updates the value of highlightedConflict state based on the value of props
+    updateHighlightedConflict = (newHighlightedConflict: QualityConflict | null): void => {
+        // eslint-disable-next-line react/no-unused-state
+        this.setState({ highlightedConflict: newHighlightedConflict });
+    };
```

Comment:
Could you once again clarify why we need this method? I dont see any applications, eslint tells us its unused.

Reply:
The value of `highlightedConflict`, which is passed as `newHighlightedConflict` in this function, is a props value. The value the `FrameTag` component sees is the state value, due to which if we go without updating the state in that case, `FrameTag` performs the logic over the last value of `highlightedConflict` corresponding to the shape type.

Reply:
Hello a gentle reminder to this 😅

Reply:
@Viditagarwal7479 , Kirill is OOO. He will come back tomorrow.

Reply:
Hi @Viditagarwal7479,
I dont think we need to update anything here. If I just comment the call of `this.updateHighlightedConflict(highlightedConflict);` the patch works anyway. 

There is a small misunderstanding of how data flow works.

Yes, the `highlightedConflict` is a props value in a `canvas-wrapper` but its not "external" prop, we get it from the global application state. We use the same `mapStateFromProps` to get it. So when the `frame-tags` uses `mapStateFromProps` it gets the same `highlightedConflict`.

So the highlighting pipeline is:
User hovers the conflict -> `issues-list` component dispatches the conflictValue, it is stored in global state -> the components which use `mapStateToProps` based on `highlightedConflicts` gets updated (in our case its `canvas-wrapper` and `frame-conflicts`) -> each one uses its own logic to update.
As you can see the conflict is stored in one place so I belive there is no outdated data.

Reply:
Hello @klakhov , 
Thank you for the detailed clarification; on top of my incorrect observation, I didn't have the look above that it's passing the state value itself as the props value. On making the change accordingly, to my surprise, as you mentioned, the highlight pipeline is working correctly.